### DISCUSSION TITLE
Can models be saved / loaded?

### DISCUSSION THREAD

**Discussion by jbone** [online-ml/river#464] [Q&A] [answered]
Hope this is the correct place for this.

Once a model has been “trained” (run over a sufficient amount of online data) can it be “saved” (persisted to a file) and subsequently “loaded” (reconstituted from a file with learned data intact) in a subsequent run / process?

thanks!

**Comment by MaxHalford** [comment, marked as answer]
Hey! Yes you can! There's an example in the [FAQ](https://riverml.xyz/latest/faq/faq/):

```py
>>> from river import ensemble
import pickle

model = ensemble.AdaptiveRandomForestClassifier()

with open('model.pkl', 'wb') as f:
    pickle.dump(model, f)
 
with open('model.pkl', 'rb') as f:
    model = pickle.load(f)
```

Each model is unit tested to make sure that it can be pickled. When you load the model, it will be in the exact same state as when you saved it.

**Comment by jbone** [comment]
Thanks!  (Was that there before and I just missed it?  If so apologies!)

jb

**Reply by MaxHalford** [reply]
It was, but don't worry :)
